prusti-contracts/src/lib.rs

```diff
@@ -84,8 +98,59 @@ mod private {
     /// A macro for defining a predicate using prusti expression syntax instead
     /// of just Rust expressions.
     pub use prusti_contracts_internal::predicate;
-}
 
+    /// A sequence type
+    #[non_exhaustive]
+    #[derive(PartialEq, Eq, Copy, Clone)]
+    pub struct Seq<T: Copy> {
+        _phantom: core::marker::PhantomData<T>,
+    }
+
+    impl<T: Copy> Seq<T> {
+        #[pure]
+        pub fn empty() -> Self {
+            panic!()
+        }
+        #[pure]
+        pub fn single(_: T) -> Self {
+            panic!()
+        }
+        #[pure]
+        pub fn concat(_: Self, _: Self) -> Self {
+            panic!()
+        }
+        #[pure]
+        pub fn lookup(self, index: usize) -> T {
+            panic!()
+        }
+    }
+
+    /// A map type
+    #[non_exhaustive]
+    #[derive(PartialEq, Eq, Copy, Clone)]
+    pub struct Map<K, V> {
+        _key_phantom: core::marker::PhantomData<K>,
+        _val_phantom: core::marker::PhantomData<V>,
+    }
+
+    impl<K, V> Map<K, V> {
```

Comment:
`Index<K>` would be useful here as well.

Reply:
Let's do this in a follow-up PR to avoid merge conflicts with modifications made by other students.